### ISSUE TITLE
Create pypiper pipeline and make it compatible with looper

### ISSUE THREAD

**Issue by oddodaoddo** [databio/episb-provider#47]
Pipeline to produce bed file analytics as json files one per bed file.

Subtask of #46

**Comment by nsheff** [comment]
use: https://github.com/databio/episb-provider/blob/master/episb-api/regionstat.R

**Comment by oddodaoddo** [comment]
Yup, of course, I sent you a whole email asking about the summary statistics and I completely ignored the last lines of that script. Thanks!

**Comment by oddodaoddo** [comment]
@nsheff I am not much of a GenomicDistributions/GenomicRanges guy (first time I see this) :blush: but I think the way it should work is the R script regionstats.R should accept a bed filename (full path) and create a GRanges object out of it (for this I use rtracklayer library) and then feed it to GenomicDistributions for further processing. Is this correct or am I misunderstanding the whole way this will interface with pypiper?

https://github.com/databio/episb-provider/commit/291f0165fe756f2c27e55c6b6961fdd877199691

The pypiper python code is incomplete since I am trying to get the R script to work properly first.

In any case, I created an Issue-47 branch which details the changes. When I run the script on a bed file, I get an error:

Loading required package: GenomeInfoDb
Saving 7 x 7 in image
Error: Faceting variables must have at least one value
Execution halted

Could you offer any insight?

Also, what .bed files can I test the script on?

Thanks!

**Comment by nsheff** [comment]
>  I think the way it should work is the R script regionstats.R should accept a bed filename (full path) and create a GRanges object out of it (for this I use rtracklayer library) and then feed it to GenomicDistributions for further processing. Is this correc

yes. exactly.

> Could you offer any insight?

I don't see anything directly here, sorry... Probably something wrong with the data objects or something. try running it in R interactively and see what it's doing and make sure you're giving it objects that make sense.

> Also, what .bed files can I test the script on?

you can use the same ones you had before; anything from [databio.org/regiondb](https://databio.org/regiondb) for example. eventually we want to make these into a PEP so looper can run it on each file in the database.

**Comment by nsheff** [comment]
I updated the regionstat R with some alternative stuff and example:

https://github.com/databio/episb-provider/blob/dev/episb-api/regionstat.R

**Comment by oddodaoddo** [comment]
> I updated the regionstat R with some alternative stuff and example:
> 
> https://github.com/databio/episb-provider/blob/dev/episb-api/regionstat.R

A few questions:

1) Where do I find files such as /ext/qumulo/resources/regions/LOLARoadmap/hg38/roadmap_epigenomics/regions/E125-DNase.macs2.narrowPeak (I am developing locally so I need to download a copy of whatever contains this file, for testing purposes) - I already got LOLA Core db but this file seems to be in a different collection
2) Is the regionstat.R script only supposed to run on hg38 or is that too an input parameter?

**Comment by nsheff** [comment]
> Where do I find files 

that's on the qumulo file system, which you should have access to. it's mounted on rivanna and I think also on dcos, somewhere... but anyway, the file is just an example, use any bed file.

> only supposed to run on hg38 or is that too an input parameter

that should perhaps eventually become an input parameter. if we can get this working for hg38 I will be happy.

**Comment by oddodaoddo** [comment]
Ok. I am having trouble with this portion of the regionstat.R script:

```	
        gcvec = calcGCContentRef(query, "hg38")
	g = plotGCContent(gcvec)
	ggplot2::ggsave(paste0(outfolder, fileid, "_gccontent.png"), g)
```

There does not seem to be a function called calcGCContentRef in GenomicDistributions, at least not within http://code.databio.org/GenomicDistributions/reference/index.html

My R is a bit rusty, I just finished Coursera's "Introduction to R" course but am still not great at searching for this stuff. Thanks!

**Comment by nsheff** [comment]
Oh... maybe I haven't written that function yet... it might have just been an example...

**Comment by nsheff** [comment]
looks like that's in the dev version but haven't merged it yet: 

https://github.com/databio/GenomicDistributions/commit/4acee8a044be13fc1c3e3ed61759345bb43ae8a5
